src/hooks/useCampaignManagement.ts

```diff
@@ -0,0 +1,289 @@
+              `"${uploadedFileInfo.filename}" was already in your campaign. No new snippets were generated.`
+            );
+          } else if (
+            result.message?.includes("Generated") &&
+            result.message.includes("snippets")
+          ) {
+            // Extract the number of snippets from the message
+            const snippetMatch = result.message.match(
+              /Generated (\d+) snippets/
+            );
+            const snippetCount = snippetMatch ? snippetMatch[1] : "some";
+            onSendNotification(
+              `"${uploadedFileInfo.filename}" has been added to my campaign and ${snippetCount} snippets were generated. Please show me these snippets so I can review and approve them.`
+            );
+          } else if (result.message?.includes("No snippets were generated")) {
+            onSendNotification(
+              `"${uploadedFileInfo.filename}" has been added to my campaign. No snippets were generated from this resource.`
+            );
+          } else {
+            onSendNotification(
+              `"${uploadedFileInfo.filename}" has been added to my campaign. The document is now being processed to extract game-ready content.`
+            );
+          }
+        }
+      } catch (error) {
+        console.error("Failed to add file to campaign:", error);
+      }
+    },
+    [onSendNotification]
+  );
+
+  const handleCreateCampaignForFile = useCallback(
+    async (uploadedFileInfo: { filename: string; fileKey: string } | null) => {
+      if (!campaignName.trim() || !uploadedFileInfo) return;
+
+      try {
+        const jwt = getStoredJwt();
+        if (!jwt) {
+          console.error("No JWT token available");
+          return;
+        }
+
+        // First create the campaign
+        const createResponse = await authenticatedFetchWithExpiration(
+          API_CONFIG.buildUrl(API_CONFIG.ENDPOINTS.CAMPAIGNS.CREATE),
+          {
+            method: "POST",
+            jwt,
+            body: JSON.stringify({
+              name: campaignName,
+              description: campaignDescription,
+            }),
+            headers: {
+              "Content-Type": "application/json",
+            },
+          }
+        );
+
+        if (!createResponse.response.ok) {
+          const errorText = await createResponse.response.text();
+          throw new Error(`Failed to create campaign: ${errorText}`);
+        }
+
+        const result = (await createResponse.response.json()) as {
+          campaign: Campaign;
+        };
+        const newCampaign = result.campaign;
+
+        // Emit campaign created event
+        emit({
+          type: EVENT_TYPES.CAMPAIGN.CREATED,
+          campaignId: newCampaign.campaignId,
+          campaignName: campaignName,
+          source: "useCampaignManagement",
+        } as CampaignEvent);
+
+        // Then add the file to the new campaign
+        await handleAddFileToCampaign(newCampaign.campaignId, uploadedFileInfo);
```

Comment:
either we should emit after we add or we should have a separate event for add, i think